nuxeo-features/nuxeo-platform-csv-export/src/main/java/org/nuxeo/ecm/platform/csv/export/validation/CSVExportValidation.java

```diff
@@ -0,0 +1,113 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Funsho David
+ */
+
+package org.nuxeo.ecm.platform.csv.export.validation;
+
+import static org.nuxeo.ecm.core.bulk.BulkActionValidationHelper.invalidParameterMessage;
+import static org.nuxeo.ecm.core.bulk.BulkActionValidationHelper.unknownParameterMessage;
+import static org.nuxeo.ecm.core.bulk.BulkActionValidationHelper.unknownValueMessage;
+import static org.nuxeo.ecm.core.bulk.action.computation.SortBlob.SORT_PARAMETER;
+import static org.nuxeo.ecm.core.bulk.action.computation.ZipBlob.ZIP_PARAMETER;
+import static org.nuxeo.ecm.platform.csv.export.computation.CSVProjectionComputation.PARAM_LANG;
+import static org.nuxeo.ecm.platform.csv.export.computation.CSVProjectionComputation.PARAM_SCHEMAS;
+import static org.nuxeo.ecm.platform.csv.export.computation.CSVProjectionComputation.PARAM_XPATHS;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import org.nuxeo.ecm.core.bulk.BulkActionValidation;
+import org.nuxeo.ecm.core.bulk.message.BulkCommand;
+import org.nuxeo.ecm.core.schema.SchemaManager;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * @since 10.10
+ */
+public class CSVExportValidation implements BulkActionValidation {
+
+    @Override
+    public void validate(BulkCommand command) throws IllegalArgumentException {
+
+        // Check for unknown parameter
+        Map<String, Serializable> params = command.getParams();
+        List<String> csvExportParams = Arrays.asList(SORT_PARAMETER, ZIP_PARAMETER, PARAM_LANG, PARAM_SCHEMAS,
+                PARAM_XPATHS);
+        for (String param : params.keySet()) {
+            if (!csvExportParams.contains(param)) {
+                throw new IllegalArgumentException(unknownParameterMessage(param, command));
+            }
+        }
+
+        // Check sort and zip parameters
+        Serializable sort = command.getParam(SORT_PARAMETER);
+        Serializable zip = command.getParam(ZIP_PARAMETER);
+        if (sort != null && !Boolean.TRUE.equals(sort) && !Boolean.FALSE.equals(sort)) {
+            throw new IllegalArgumentException(invalidParameterMessage(SORT_PARAMETER, command));
+        }
+        if (zip != null && !Boolean.TRUE.equals(zip) && !Boolean.FALSE.equals(zip)) {
+            throw new IllegalArgumentException(invalidParameterMessage(ZIP_PARAMETER, command));
```

Comment:
I think there's a lot of boilerplate in the various `BulkActionValidation` implementations, and it's worth adding an `AbstractBulkActionValidation` with helper methods like `requireBoolean(String name)`, `requireSchema(String name)`, etc.